.travis.yml

```diff
@@ -10,4 +10,7 @@ env:
     - TOX_VERSION="tox-3.x"
 install:
     - pip install --constraint "$TOX_VERSION" .
-script: "tox"
+script:
+    - tox
+    # assure that we fail a build if it contains leftovers
+    - "bash -c 'test -z \"$(git status --porcelain | tee /dev/fd/2)\"'"
```

Comment:
Do we know why this is happening? There shouldn't be any changes in tracked files due to running tests/tox. If there are, let's figure out what's causing that, and fix that.

Reply:
@dcrosta crosta is not happening now but I am trying to prevent regressions like the ones causing this fix:
like missing build/ or dist/ to be added to .gitignore.

By running this at the end, we assure we never merge stuff that is producing unexpected lefovers.

There are also some linters that have the habbit of altering the source code when they run (like auto-fixing ones) and that is why is better to assure a failure exit code when codebase is dirty.

If you want I can take this out of this review and add it to a separate one.

Reply:
Oh, I see, this is more about producing new files, rather than changing tracked files. Yeah, for that, I wouldn't want this in here -- I think it would cause more frustration than it would solve problems -- it's expected that test runs will produce some artifacts (results files for CI integration, coverage reports, etc), and that those probably should go in gitignore, but don't necessarily have to.